pkg/build/api/v1beta1/conversion.go

```diff
@@ -264,26 +264,41 @@ func init() {
 				ref.Registry = ""
 				out.ImageTag = ref.String()
 			}
+			if (out.To != nil || len(in.DockerImageReference) > 0) && len(out.Tag) == 0 {
+				out.Tag = imageapi.DefaultImageTag
```

Comment:
why set out.Tag if it's an in.DockerImageReference?  out.Tag is only used when out.To!=nil.